lib/AST/ASTMangler.cpp

```diff
@@ -369,51 +369,86 @@ std::string ASTMangler::mangleReabstractionThunkHelper(
   return finalize();
 }
 
+// SWIFT_ENABLE_TENSORFLOW
+/// Get a `NodePointer` representing an autodiff associated function for the
+/// given original function mangled name, associated function node kind,
+/// and parameter indices.
+static NodePointer getMangledAutoDiffAssociatedFunctionNode(
```

Comment:
Note for the curious: this logic was adapted from [`SpecializationMangler::finalize`](https://github.com/apple/swift/blob/4df23c8e12f718b47729d5546a0fe188eab1d490/lib/SILOptimizer/Utils/SpecializationMangler.cpp#L49).